src/Templates/ArchiveProductTemplatesCompatibility.php

```diff
@@ -202,20 +230,28 @@ protected function set_hook_data() {
 				),
 			),
 			'woocommerce_before_shop_loop'            => array(
-				'block_name' => 'core/post-template',
-				'position'   => 'before',
-				'hooked'     => array(
+				'block_name'            => '',
+				'position'              => 'before',
+				'hooked'                => array(
 					'woocommerce_output_all_notices' => 10,
 					'woocommerce_result_count'       => 20,
 					'woocommerce_catalog_ordering'   => 30,
 				),
+				'not_re_addble_actions' => array(
+					'woocommerce_output_all_notices',
+					'woocommerce_result_count',
+					'woocommerce_catalog_ordering',
+				),
 			),
 			'woocommerce_after_shop_loop'             => array(
-				'block_name' => 'core/post-template',
-				'position'   => 'after',
-				'hooked'     => array(
+				'block_name'            => '',
+				'position'              => 'after',
+				'hooked'                => array(
 					'woocommerce_pagination' => 10,
 				),
+				'not_re_addble_actions' => array(
```

Comment:
I think a better name for this key would be `removed_actions` and document at the top of the function what this key means for additional context:

"Actions that we do not want to re-add after they have been removed because..."